### ISSUE TITLE
virtualization resource should detect vmware

### ISSUE THREAD

**Issue by mlcooper** [inspec/inspec#2509]
### Description

This is an enhancement request for the inspec virtualization resource to detect vmware.

### InSpec and Platform Version

1.49.2

### Replication Case
```
describe virtualization do
  its('system') { should include 'vmware' }
  its('role') { should include 'guest' }
end
```
results in the following error on a known good VMware guest (RHEL 7.4)
```
× Virtualization Detection system should eq "vmware"
expected: "vmware"
got: nil
(compared using ==)
× Virtualization Detection role should eq "guest"
expected: "guest"
got: nil
(compared using ==)
```

**Comment by adamleff** [comment]
Thanks for your feature request, @mlcooper.

**Comment by PopPopDevOps** [comment]
I think this could be as easy as the following:

```ruby
    # Detect VMware systems via /sys
    # guests will have the Virtual Machine Communication Interface
    def detect_vmware_from_sys
      return false unless inspec.file('/sys/devices/virtual/dmi/id/sys_vendor').exist?
      @virtualization_data[:system] = 'vmware'
        @virtualization_data[:role] = 'guest'
    end
```

(@zenspider: edited formatting)

**Comment by clintoncwolfe** [comment]
#4954 implemented support for vmware detection, released in 4.18.104, closing
